src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    static final int INITIAL_SIZE = 10;
+    static final double growIndex = 1.5;
+    private T[] elements;
+    private int size;
```

Comment:
It is better to initialize the size field right away
```suggestion
    private int size = 0;
```